index.js

```diff
@@ -71,15 +71,17 @@ if (program.obj) {
  * searched first.
  */
 function parseObj (input) {
-  var str, out;
+  var str;
   try {
-    str = fs.readFileSync(program.obj);
+    str = fs.readFileSync(program.obj, 'utf8');
   } catch (e) {
-    return eval('(' + program.obj + ')');
+    str = program.obj;
+  }
+  try {
+    return JSON.parse(str);
+  } catch (e) {
+    return eval('(' + str + ')');
```

Comment:
Before:

```
read file  ------------------+
   | works      doesn't work |                                                  
   v                         v    doesn't work
JSON.parse --------+       eval ---------------+
   | works  doesn't| work    |                 |   
   v               |         | works           |   
output <-----------+---------+                 |   
                   |                           v   
                   +-----------------------> throw
```

What's the problem? If the object is specified as a JSON string, and if `eval` doesn't work, we didn't try `JSON.parse`. Same thing but reversed with a file: if `JSON.parse` doesn't work, we didn't try `eval`.

So, I consolidated the string generation (either reading a file or using the argument verbatim as a string), and consolidated the parsing part (`JSON.parse` or `eval`).

Now:

```
   +------ read file --------+
   | works      doesn't work |
   +-----------+-------------|
               |  
               v         works
           JSON.parse ----------> output
               |   doesn't work
               v  
             eval     ----------> output 
               |   doesn't work
               v  
             throw
```